crates/oxc_parser/src/lexer/trivia_builder.rs

```diff
@@ -201,6 +202,36 @@ impl TriviaBuilder {
     }
 }
 
+#[expect(clippy::inline_always)]
+#[inline(always)]
+fn contains_license_or_preserve_comment(s: &str) -> bool {
+    let hay = s.as_bytes();
+
+    if hay.len() < 9 {
+        return false;
+    }
+
+    let search_len = hay.len() - 8;
+
+    for i in memchr_iter(b'@', &hay[..search_len]) {
+        match hay[i + 1] {
+            b'l' => {
+                if &hay[i + 2..i + 1 + 7] == b"icense" {
+                    return true;
+                }
+            }
+            b'p' => {
+                if &hay[i + 2..i + 1 + 8] == b"reserve" {
+                    return true;
+                }
+            }
+            _ => {}
+        }
+    }
+
+    false
+}
+
 #[cfg(test)]
```

Comment:
We probably need a few more edge cases to catch all the branches (if you can think of any).